scm-ui/ui-webapp/src/admin/plugins/containers/PluginsOverview.tsx

```diff
@@ -35,89 +34,54 @@ import {
 import PluginTopActions from "../components/PluginTopActions";
 import PluginBottomActions from "../components/PluginBottomActions";
 import ExecutePendingActionModal from "../components/ExecutePendingActionModal";
 import CancelPendingActionModal from "../components/CancelPendingActionModal";
 import UpdateAllActionModal from "../components/UpdateAllActionModal";
 import ShowPendingModal from "../components/ShowPendingModal";
+import { useAvailablePlugins, useInstalledPlugins, usePendingPlugins } from "@scm-manager/ui-api";
+import PluginModal from "../components/PluginModal";
 
-type Props = WithTranslation & {
-  loading: boolean;
-  error: Error;
-  collection: PluginCollection;
-  baseUrl: string;
-  installed: boolean;
-  availablePluginsLink: string;
-  installedPluginsLink: string;
-  pendingPluginsLink: string;
-  pendingPlugins: PendingPlugins;
+export enum PluginAction {
+  INSTALL = "install",
+  UPDATE = "update",
+  UNINSTALL = "uninstall"
+}
 
-  // dispatched functions
-  fetchPluginsByLink: (link: string) => void;
-  fetchPendingPlugins: (link: string) => void;
-};
+export type PluginModalContent = {
+  plugin: Plugin;
+  action: PluginAction;
+}
 
-type State = {
-  showPendingModal: boolean;
-  showExecutePendingModal: boolean;
-  showUpdateAllModal: boolean;
-  showCancelModal: boolean;
+type Props = {
+  installed: boolean;
 };
 
-class PluginsOverview extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      showPendingModal: false,
-      showExecutePendingModal: false,
-      showUpdateAllModal: false,
-      showCancelModal: false
-    };
-  }
-
-  componentDidMount() {
-    this.fetchPlugins();
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    const { installed } = this.props;
-    if (prevProps.installed !== installed) {
-      this.fetchPlugins();
-    }
-  }
-
-  fetchPlugins = () => {
-    const {
-      installed,
-      fetchPluginsByLink,
-      availablePluginsLink,
-      installedPluginsLink,
-      pendingPluginsLink,
-      fetchPendingPlugins
-    } = this.props;
-    fetchPluginsByLink(installed ? installedPluginsLink : availablePluginsLink);
-    if (pendingPluginsLink) {
-      fetchPendingPlugins(pendingPluginsLink);
-    }
-  };
-
-  renderHeader = (actions: React.ReactNode) => {
-    const { installed, t } = this.props;
+const PluginsOverview: FC<Props> = ({ installed }) => {
+  const [t] = useTranslation("admin");
+  const {
+    data: availablePlugins,
+    isLoading: isLoadingAvailablePlugins,
+    error: availablePluginsError
+  } = useAvailablePlugins({ enabled: !installed });
+  const {
+    data: installedPlugins,
+    isLoading: isLoadingInstalledPlugins,
+    error: installedPluginsError
+  } = useInstalledPlugins({ enabled: installed });
+  const { data: pendingPlugins, isLoading: isLoadingPendingPlugins, error: pendingPluginsError } = usePendingPlugins();
+  const [showPendingModal, setShowPendingModal] = useState<boolean>();
```

Comment:
```suggestion
  const [showPendingModal, setShowPendingModal] = useState(false);
```

I think this is easier to understand.